tools/istio-nftables/pkg/capture/run.go

```diff
@@ -0,0 +1,1026 @@
+	}
+	ipv6Ranges := NetworkRange{}
+	ipv4Ranges := NetworkRange{}
+	for _, ipRange := range split(cidrList) {
+		ipp, err := netip.ParsePrefix(ipRange)
+		if err != nil {
+			_, err = fmt.Fprintf(os.Stderr, "Ignoring error for bug compatibility with istio-nftables: %s\n", err.Error())
+			if err != nil {
+				return ipv4Ranges, ipv6Ranges, err
+			}
+			continue
+		}
+		if ipp.Addr().Is4() {
+			ipv4Ranges.CIDRs = append(ipv4Ranges.CIDRs, ipp)
+			if ipp.Addr().IsLoopback() {
+				ipv4Ranges.HasLoopBackIP = true
+			}
+		} else {
+			ipv6Ranges.CIDRs = append(ipv6Ranges.CIDRs, ipp)
+			if ipp.Addr().IsLoopback() {
+				ipv6Ranges.HasLoopBackIP = true
+			}
+		}
+	}
+	return ipv4Ranges, ipv6Ranges, nil
+}
+
+// logConfig prints out the current environment variable values and the loaded config.
+// This is useful for debugging to verify which settings are being used.
+func (cfg *NftablesConfigurator) logConfig() {
+	// Dump out our environment for debugging purposes.
+	var b strings.Builder
+	b.WriteString(fmt.Sprintf("ENVOY_PORT=%s\n", os.Getenv("ENVOY_PORT")))
+	b.WriteString(fmt.Sprintf("INBOUND_CAPTURE_PORT=%s\n", os.Getenv("INBOUND_CAPTURE_PORT")))
+	b.WriteString(fmt.Sprintf("ISTIO_INBOUND_INTERCEPTION_MODE=%s\n", os.Getenv("ISTIO_INBOUND_INTERCEPTION_MODE")))
+	b.WriteString(fmt.Sprintf("ISTIO_INBOUND_TPROXY_ROUTE_TABLE=%s\n", os.Getenv("ISTIO_INBOUND_TPROXY_ROUTE_TABLE")))
+	b.WriteString(fmt.Sprintf("ISTIO_INBOUND_PORTS=%s\n", os.Getenv("ISTIO_INBOUND_PORTS")))
+	b.WriteString(fmt.Sprintf("ISTIO_OUTBOUND_PORTS=%s\n", os.Getenv("ISTIO_OUTBOUND_PORTS")))
+	b.WriteString(fmt.Sprintf("ISTIO_LOCAL_EXCLUDE_PORTS=%s\n", os.Getenv("ISTIO_LOCAL_EXCLUDE_PORTS")))
+	b.WriteString(fmt.Sprintf("ISTIO_EXCLUDE_INTERFACES=%s\n", os.Getenv("ISTIO_EXCLUDE_INTERFACES")))
+	b.WriteString(fmt.Sprintf("ISTIO_SERVICE_CIDR=%s\n", os.Getenv("ISTIO_SERVICE_CIDR")))
+	b.WriteString(fmt.Sprintf("ISTIO_SERVICE_EXCLUDE_CIDR=%s\n", os.Getenv("ISTIO_SERVICE_EXCLUDE_CIDR")))
+	b.WriteString(fmt.Sprintf("ISTIO_META_DNS_CAPTURE=%s\n", os.Getenv("ISTIO_META_DNS_CAPTURE")))
+	b.WriteString(fmt.Sprintf("INVALID_DROP=%s\n", os.Getenv("INVALID_DROP")))
+	log.Infof("Istio nftables environment:\n%s", b.String())
+	cfg.cfg.Print()
+}
+
+// handleInboundPortsInclude sets up rules to redirect inbound TCP traffic to Envoy.
+// If TPROXY is enabled, it adds rules to mark packets and route them through mangle tables.
+// It also handles excluded ports and loopback exceptions, based on the config.
+// The goal here is to make sure incoming traffic reaches Envoy before going to the actual app.
+func (cfg *NftablesConfigurator) handleInboundPortsInclude() {
+	// Handling of inbound ports. Traffic will be redirected to Envoy, which will process and forward
+	// to the local service. If not set, no inbound port will be intercepted by istio nftablesOrFail.
+	var table string
+	if cfg.cfg.InboundPortsInclude != "" {
+		if cfg.cfg.InboundInterceptionMode == "TPROXY" {
```

Comment:
Please create a separate function for tproxy for readability

Reply:
Done as part of the following commit - https://github.com/istio/istio/pull/56487/commits/82eee5834024b1e2d186a8cea2c97726a3535d96